src/core/src/model.cpp

```diff
@@ -470,6 +470,18 @@ int64_t ov::Model::get_result_index(const Output<const Node>& value) const {
     return -1;
 }
 
+int64_t ov::Model::get_result_index(const ov::op::v0::Result& result) const {
+    int64_t pos = 0;
+    auto result_ptr = result.shared_from_this();
+    for (auto r : get_results()) {
+        if (r == result_ptr) {
+            return pos;
+        }
+        pos++;
+    }
+    return -1;
```

Comment:
It can be solve on pybind only, 

But if kept then it can just call current implementation, and add unit test in `src/core/tests/model.cpp` for new member.

```suggestion
    return get_result_index(result.get_default_output());
```

Reply:
have added the unit tests in core/tests/model.cpp file